pilot/pkg/model/service.go

```diff
@@ -612,15 +612,21 @@ func ParseServiceKey(s string) (hostname Hostname, ports PortList, labels Labels
 // BuildSubsetKey generates a unique string referencing service instances for a given service name, a subset and a port.
 // The proxy queries Pilot with this key to obtain the list of instances in a subset.
 func BuildSubsetKey(direction TrafficDirection, subsetName string, hostname Hostname, port *Port) string {
-	return fmt.Sprintf("%s|%s|%s|%s", direction, port.Name, subsetName, hostname)
+	if subsetName == "" {
+		subsetName = "_default_"
```

Comment:
This should be a const (can be above this function since it's localized).

Reply:
On second thought, maybe better to put it at the top with a comment since this string will be popping up everywhere now.